src/controllers/scripting/legacy/controllerscriptenginelegacy.cpp

```diff
@@ -192,8 +192,8 @@ bool ControllerScriptEngineLegacy::handleIncomingData(const QByteArray& data) {
             static_cast<uint>(data.size()),
     };
 
-    for (const QJSValue& function : std::as_const(m_incomingDataFunctions)) {
-        ControllerScriptEngineBase::executeFunction(function, args);
+    for (auto&& function : m_incomingDataFunctions) {
```

Comment:
why the `&&`? does/could that cause a move out of the container?

Reply:
No, in this case this is the "universal reference"  or a "forwarding reference" that reliable hat the type of *iterator. 
https://stackoverflow.com/questions/13230480/what-does-auto-tell-us
This comes from the pattern to use `const auto&` and `auto&&` as an universal rule of thump that is always correct. 
In this case we have a QList where *iterator is QJSValue&
https://en.cppreference.com/w/cpp/language/range-for dos not even mention the `auto&` case
I can change that, but how do we want to continue if Mixxx in general?

Reply:
> how do we want to continue in Mixxx in general?

I'm not sure, if in doubt, consult the cpp core guidelines. The problem right now for me is that the topic is highly intertwined with overload resolution and value categories so I've not managed to wrap my head around the topic yet. 

keep it as it for now. I'm sure there won't be any move since the value is named.